mlflow/utils/search_utils.py

```diff
@@ -54,7 +52,7 @@ class SearchUtils(object):
     # Registered Models Constants
     ORDER_BY_KEY_TIMESTAMP = "timestamp"
     ORDER_BY_KEY_LAST_UPDATED_TIMESTAMP = "last_updated_timestamp"
-    ORDER_BY_KEY_MODEL_NAME = SqlRegisteredModel.name.key
```

Comment:
Can we do 2 things.

1. Move some of the column name constants to a lower level utils or stores package as `column_names.py` and share it for search utils and also SQL Alchemy stuff.
2. These this columns as names on SqlAlchemyStore data models using [name param in constructor](https://docs.sqlalchemy.org/en/13/core/metadata.html#sqlalchemy.schema.Column.__init__). For example, 

```
class SqlRegisteredModel(Base):
    __tablename__ = "registered_models"

    name = Column(String(256), unique=True, nullable=False)

    creation_time = Column(BigInteger, default=lambda: int(time.time() * 1000))

    ...
```

becomes

```
from mlflow.stores.db.column_names import *

class SqlRegisteredModel(Base):
    __tablename__ = "registered_models"

    name = Column(REGISTERED_MODEL_NAME, String(256), unique=True, nullable=False)

    creation_time = Column(REGISTERED_MODEL_CREATION_TIME, BigInteger, default=lambda: int(time.time() * 1000))
```

I think it would involve some work for registry and tracking data models and changes to test code. Also, I expect some changes to search utils. Does not need to block the release, but would be awesome to DRY these constants.